### PR TITLE
Improve JOB_POD variable naming + improve doc about memory management

### PR DESCRIPTION
**Pull request by alafanechere**
## What
* Some env var have `POD` in their name but are not specific to Kubernetes deployment: I suggest removing `POD` from their name.
* Our [Scaling Airbyte documentation](https://docs.airbyte.io/operator-guides/scaling-airbyte) is not mentioning that we enable users to customize memory requests and limits for sync jobs.

## How
* Rename `JOB_POD_*` to `JOB_*` env var when they are not strictly related to K8S deployments
* Update the Scaling airbyte documentation to mention `JOB_MAIN_CONTAINER_MEMORY_REQUEST` and `JOB_MAIN_CONTAINER_MEMORY_LIMIT` variable.

## Recommended reading order
1. Double check my variable renaming is correct
2. Check if method names are consistent.
3. Check if the doc update is clear enough. 

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -47,9 +47,9 @@ SUBMITTER_NUM_THREADS=10
 
 # Job container images
 # Usually you should not need to set these, they have defaults already set
-JOB_POD_SOCAT_IMAGE=
-JOB_POD_BUSYBOX_IMAGE=
-JOB_POD_CURL_IMAGE=
+JOB_SOCAT_IMAGE=
```
Comment (davinchia) [inline .env:50]: these are only specific to kube, so we should keep pod in the name

File: .env [hunk 2]
```diff
@@ -47,9 +47,9 @@ SUBMITTER_NUM_THREADS=10
...
+JOB_BUSYBOX_IMAGE=
+JOB_CURL_IMAGE=
```
Comment (davinchia) [inline .env:52]: same as SOCAT

**Review by davinchia — COMMENTED** [review]
Thanks for taking this on. Agree leaving pod in the name was confusing.

Instead of only removing pod, what about introducing a convention where Kube only Job variables have the `JOB_KUBE_` prefix? e.g. `JOB_KUBE_NAMESPACE`, `JOB_KUBE_SOCAT_IMAGE` etc. It would help differentiate between general job variables, and job variables that only apply to K8s.

**Comment by alafanechere** [comment]
> Thanks for taking this on. Agree leaving pod in the name was confusing.
> 
> Instead of only removing pod, what about introducing a convention where Kube only Job variables have the `JOB_KUBE_` prefix? e.g. `JOB_KUBE_NAMESPACE`, `JOB_KUBE_SOCAT_IMAGE` etc. It would help differentiate between general job variables, and job variables that only apply to K8s.

Agreed 🎯  Will do.

File: airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java [hunk 3]
```diff
@@ -56,12 +56,12 @@
   public static final String LOG_LEVEL = "LOG_LEVEL";
   public static final String S3_PATH_STYLE_ACCESS = "S3_PATH_STYLE_ACCESS";
   public static final String WEBAPP_URL = "WEBAPP_URL";
-  public static final String JOB_POD_MAIN_CONTAINER_IMAGE_PULL_POLICY = "JOB_POD_MAIN_CONTAINER_IMAGE_PULL_POLICY";
-  public static final String JOB_POD_TOLERATIONS = "JOB_POD_TOLERATIONS";
-  public static final String JOB_POD_NODE_SELECTORS = "JOB_POD_NODE_SELECTORS";
-  public static final String JOB_POD_SOCAT_IMAGE = "JOB_POD_SOCAT_IMAGE";
-  public static final String JOB_POD_BUSYBOX_IMAGE = "JOB_POD_BUSYBOX_IMAGE";
-  public static final String JOB_POD_CURL_IMAGE = "JOB_POD_CURL_IMAGE";
+  public static final String JOB_MAIN_CONTAINER_IMAGE_PULL_POLICY = "JOB_MAIN_CONTAINER_IMAGE_PULL_POLICY";
```
Comment (davinchia) [inline airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java:59]: should this also be changed to JOB_KUBE?

File: airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java [hunk 4]
```diff
@@ -375,7 +375,7 @@ public int getSyncJobMaxTimeoutDays() {
    */
   @Override
   public List<TolerationPOJO> getJobPodTolerations() {
```
Comment (davinchia) [inline airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java:377]: please rename the methods too

File: airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java [hunk 5]
```diff
@@ -73,14 +73,14 @@
...
-  public static final String JOB_POD_MAIN_CONTAINER_CPU_LIMIT = "JOB_POD_MAIN_CONTAINER_CPU_LIMIT";
-  public static final String JOB_POD_MAIN_CONTAINER_MEMORY_REQUEST = "JOB_POD_MAIN_CONTAINER_MEMORY_REQUEST";
-  public static final String JOB_POD_MAIN_CONTAINER_MEMORY_LIMIT = "JOB_POD_MAIN_CONTAINER_MEMORY_LIMIT";
+  public static final String JOB_MAIN_CONTAINER_CPU_REQUEST = "JOB_MAIN_CONTAINER_CPU_REQUEST";
+  public static final String JOB_MAIN_CONTAINER_CPU_LIMIT = "JOB_MAIN_CONTAINER_CPU_LIMIT";
+  public static final String JOB_MAIN_CONTAINER_MEMORY_REQUEST = "JOB_MAIN_CONTAINER_MEMORY_REQUEST";
+  public static final String JOB_MAIN_CONTAINER_MEMORY_LIMIT = "JOB_MAIN_CONTAINER_MEMORY_LIMIT";
   private static final String SECRET_PERSISTENCE = "SECRET_PERSISTENCE";
-  public static final String JOB_POD_MAIN_CONTAINER_IMAGE_PULL_SECRET = "JOB_POD_MAIN_CONTAINER_IMAGE_PULL_SECRET";
+  public static final String JOB_MAIN_CONTAINER_IMAGE_PULL_SECRET = "JOB_MAIN_CONTAINER_IMAGE_PULL_SECRET";
```
Comment (davinchia) [inline airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java:83]: this is also only for Kube

File: airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java [hunk 6]
```diff
@@ -73,14 +73,14 @@
   public static final String MAX_SYNC_WORKERS = "MAX_SYNC_WORKERS";
   private static final String TEMPORAL_HOST = "TEMPORAL_HOST";
   private static final String TEMPORAL_WORKER_PORTS = "TEMPORAL_WORKER_PORTS";
-  private static final String JOB_POD_KUBE_NAMESPACE = "JOB_POD_KUBE_NAMESPACE";
+  private static final String JOB_KUBE_KUBE_NAMESPACE = "JOB_KUBE_KUBE_NAMESPACE";
```
Comment (davinchia) [inline airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java:76]: I'm thinking `JOB_KUBE_NAMESPACE` is clear enough. wdyt?

**Review by davinchia — COMMENTED** [review]
Some more renames!

Please also make sure the methods match the variable names.

**Review by davinchia — APPROVED** [review]
Thanks!

**Pull request merged**